src/smt/assertions.cpp

```diff
@@ -138,9 +139,12 @@ context::CDList<Node>* Assertions::getAssertionList()
 {
   return d_produceAssertions ? &d_assertionList : nullptr;
 }
+context::CDList<Node>* Assertions::getAssertionListDefinitions()
+{
+  return d_produceAssertions ? &d_assertionListDefs : nullptr;
+}
```

Comment:
Can we please change this pattern?

Apart from the interesting semantics of the return value (that is not even mentioned in the function documentation), it is completely unnecessary: there is only a single place where `getAssertionList()` is called and we do not immediately assert the result to be `!= nullptr` (in `sygus_solver.cpp`); furthermore, if `!d_produceAssertions` the list is empty anyway.
Simply return the list (as a `const &`), get rid of all the pointer stuff and convert the `nullptr` check in the sygus solver by a `.empty()`.